controllers/repository_controller.go

```diff
@@ -143,6 +144,9 @@ func (r *RepositoryReconciler) checkInitial(ctx context.Context, logger logr.Log
 
 	if v, ok := instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel]; !ok || v != instanceDeepCopy.Spec.RepositoryType {
 		instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel] = instanceDeepCopy.Spec.RepositoryType
+		if repoType, err := r.getRepoType(instanceDeepCopy); err == nil && repoType != "" {
+			instanceDeepCopy.Labels[corev1alpha1.RepositoryTypeLabel] = repoType
```

Comment:
@zqq454224016 
You can define `GetRepoType` in `api/v1alpha1/repository.go`

```golang

type RepositoryType string

const (
	RepositoryTypeOCI         RepositoryType = "oci"
	RepositoryTypeChartmuseum RepositoryType = "chartmuseum"
	RepositoryUnknown         RepositoryType = "unknown"
)

func (r *Repository) GetRepoType() RepositoryType {
	if r.IsOCI() {
		return RepositoryTypeOCI
	}
	if r.IsChartmuseum() {
		return RepositoryTypeChartmuseum
	}
	return RepositoryUnknown
}

// IsOCI determines whether the repository is to be treated as an OCI repo
func (r *Repository) IsOCI() bool {
	return registry.IsOCI(r.Spec.URL)
}

// IsChartmuseum determines whether the repository is to be treated as a chartmuseum
func (r *Repository) IsChartmuseum() bool {
	// reference apis: https://github.com/helm/chartmuseum#chart-manipulation
	req, err := http.NewRequest("POST", r.Spec.URL+"/api/charts", nil)
	if err != nil {
		return false
	}

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil || resp == nil {
		return false
	}
	defer resp.Body.Close()

	// when api exists, it is `chartmuseum`
	return resp.StatusCode != http.StatusNotFound
}

```

Reply:
@zqq454224016 @Abirdcfly @0xff-dev 
<img width="576" alt="image" src="https://github.com/kubebb/core/assets/30621793/b6456498-7f30-470b-8d5a-845d5e1e3ba1">